packages/flutter_tools/lib/src/commands/emulators.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/common.dart';
+import '../base/platform.dart';
+import '../base/utils.dart';
+import '../doctor.dart';
+import '../emulator.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class EmulatorsCommand extends FlutterCommand {
+  EmulatorsCommand() {
+    argParser.addOption('start',
+        help: 'The full or partial ID of the emulator to start.');
+  }
+
+  @override
+  final String name = 'emulators';
+
+  @override
+  final String description = 'List all available emulators.';
+
+  @override
+  Future<Null> runCommand() async {
+    if (doctor.workflows.every((Workflow w) => !w.canListEmulators)) {
+      throwToolExit(
+          'Unable to find any emulator sources. Please ensure you have some\n'
+          'Android AVD images ' + (platform.isMacOS ? 'or an iOS Simulator ' : '')
+          + 'available.',
+          exitCode: 1);
+    }
+
+    if (argResults.wasParsed('start')) {
+      await _startEmulator(argResults['start']);
+    } else {
+      await _listEmulators();
+    }
+  }
+
+  Future<Null> _startEmulator(String id) async {
+    final List<Emulator> emulators =
+        await emulatorManager.getEmulatorsById(id).toList();
+
+    if (emulators.isEmpty) {
+      printStatus("No emulator found that matches the ID '$id'.");
+    } else if (emulators.length > 1) {
+      printStatus("More than one emulator matches the ID '$id':\n");
+      Emulator.printEmulators(emulators);
+    } else {
+      emulators.first.launch();
+    }
+  }
+
+  Future<Null> _listEmulators() async {
+    final List<Emulator> emulators =
+        await emulatorManager.getAllAvailableEmulators().toList();
+
+    if (emulators.isEmpty) {
+      printStatus('No emulators available.\n\n'
+          // TODO: Change these when we support creation
+          // 'You may need to create images using "flutter emulators --create"\n'
+          'You may need to create one using Android Studio\n'
+          'or visit https://flutter.io/setup/ for troubleshooting tips.');
+      final List<String> diagnostics =
+          await emulatorManager.getEmulatorDiagnostics();
+      if (diagnostics.isNotEmpty) {
+        printStatus('');
+        for (String diagnostic in diagnostics) {
+          printStatus('• ${diagnostic.replaceAll('\n', '\n  ')}');
```

Comment:
I don't think we prefix lists of errors w/ bullets elsewhere. If we know these are list items, perhaps indenting would be enough; else we can just print the lines.

Reply:
This was actually copied from the devices file. We don't currently use it; I wasn't sure if to leave it in case we add some or remove. I'll remove for now, since I'm not sure we'll really have any good use for it.